test/Sentry.Hangfire.Tests/HangfireTests.cs

```diff
@@ -0,0 +1,82 @@
+using Hangfire;
+using Hangfire.MemoryStorage;
+
+namespace Sentry.Hangfire.Tests;
+
+public class HangfireTests : IClassFixture<HangfireFixture>
+{
+    private readonly HangfireFixture _fixture;
+
+    public HangfireTests(HangfireFixture hangfireFixture)
+    {
+        _fixture = hangfireFixture;
+    }
+
+    [Fact]
+    public async void ExecuteJobWithAttribute_CapturesCheckInInProgressAndOk()
+    {
+        var sentryId = SentryId.Create();
+        _fixture.Hub.CaptureCheckIn(Arg.Any<SentryCheckIn>()).Returns(sentryId);
+
+        BackgroundJob.Enqueue<TestJob>(job => job.ExecuteJobWithAttribute());
+
+        await Task.Delay(100);
```

Comment:
can we rewrite this without any delay? relying on this stuff causes tests to be flaky.
Ideally you share a signal and wait on one end and signal from the other.

You could signal from the `CaptureCheckIn` method even, and wait below on the signal with a timeout so if the test fails it doesn't hang forever:

example; https://github.com/getsentry/sentry-dotnet/blob/c19fb14d5c2a56cbe162f1ef5e30af4a8149b5fa/test/Sentry.Tests/Integrations/AppDomainAdapterTests.cs#L14-L23

Reply:
Yessir!